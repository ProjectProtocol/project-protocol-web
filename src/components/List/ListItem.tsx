import classNames from 'classnames'
import { Card, CardProps } from 'react-bootstrap'
import usePointerState from 'src/hooks/usePointerState'

export interface IListItem extends CardProps {
  onClick?: () => void
  children: React.ReactNode
  cardClasses?: string
}

export default function ListItem({
  children,
  onClick,
  cardClasses,
  ...cardProps
}: IListItem) {
  const { hover, pressActive, pointerHandlers } = usePointerState()

  return (
    <Card
      {...pointerHandlers}
      className={classNames(`pe-auto border-0 ${cardClasses}`, {
        shadow: hover,
        'shadow-sm': !hover,
        'bg-dark-subtle shadow-none': pressActive,
        'bg-white': !pressActive,
      })}
      style={{ transition: 'box-shadow 0.5s' }}
      onClick={onClick}
      role={onClick ? 'button' : ''}
      {...cardProps}
    >
      {children}
    </Card>
  )
}
