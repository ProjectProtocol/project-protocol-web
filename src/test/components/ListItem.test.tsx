import { render, screen, fireEvent } from '@testing-library/react'
import { it, expect } from 'vitest'

import ListItem from 'src/components/List/ListItem'

it('matches the snapshot', async () => {
  const r = render(
    <ListItem>
      <span>I am a child</span>
    </ListItem>,
  )

  expect(r.baseElement).toMatchSnapshot()
})

it('updates when clicked', async () => {
  const r = render(
    <ListItem data-testid="list-item">
      <span>I am a child</span>
    </ListItem>,
  )
  fireEvent.mouseDown(screen.getByTestId('list-item'))
  expect(r.baseElement).toMatchSnapshot()
})

it('updates when touched', async () => {
  const r = render(
    <ListItem data-testid="list-item">
      <span>I am a child</span>
    </ListItem>,
  )
  fireEvent.touchStart(screen.getByTestId('list-item'))
  expect(r.baseElement).toMatchSnapshot()
})

it('renders the children that are passed', async () => {
  render(
    <ListItem>
      <span>I am a child</span>
    </ListItem>,
  )
  expect(screen.getByText('I am a child')).toBeTruthy()
})
