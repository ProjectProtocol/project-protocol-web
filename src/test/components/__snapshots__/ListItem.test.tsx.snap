// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="pe-auto border-0 undefined shadow-sm bg-white card"
      role=""
      style="transition: box-shadow 0.5s;"
    >
      <span>
        I am a child
      </span>
    </div>
  </div>
</body>
`;

exports[`updates when clicked 1`] = `
<body>
  <div>
    <div
      class="pe-auto border-0 undefined shadow-sm bg-dark-subtle shadow-none card"
      data-testid="list-item"
      role=""
      style="transition: box-shadow 0.5s;"
    >
      <span>
        I am a child
      </span>
    </div>
  </div>
</body>
`;

exports[`updates when touched 1`] = `
<body>
  <div>
    <div
      class="pe-auto border-0 undefined shadow-sm bg-white card"
      data-testid="list-item"
      role=""
      style="transition: box-shadow 0.5s;"
    >
      <span>
        I am a child
      </span>
    </div>
  </div>
</body>
`;
